src/lib/date/iso-date-parser.js

```diff
@@ -0,0 +1,20 @@
+import { isDateValid } from './date-utils.js';
+
+export class IsoDateParser {
+  /**
+   * @param {string} string - The string to parse
+   * @return {Date} - The date.
+   * @throws {Error} Whenever the given string is not a valid ISO date.
+   */
+  parse (string) {
+    const regex = /[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}\.[0-9]{3}(Z|[+-][0-9]{2}:[0-9]{2})/g;
+
+    if (regex.test(string)) {
+      const date = new Date(string);
+      if (isDateValid(date)) {
+        return date;
+      }
+    }
+    throw new Error(`Invalid date "${string}"`);
+  }
+}
```

Comment:
I think I get it now that I've finished the review :sweat_smile: it might be weird to use a function for this parser when we use a class for the other.

You can ignore my comments :wink: